tests/unit/test_mesmer_x_expression.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import scipy as sp
+
+from mesmer.mesmer_x import Expression
+
+inf = float("inf")
+
+
+def test_expression_genextreme():
+
+    expression_str = (
+        "genextreme(loc=c1 + c2 * __pred1__, scale=c3 + c4 * __pred2__**2, c=c5)"
+    )
+
+    expr = Expression(expression_str, expr_name="name")
+
+    assert expr.expression == expression_str
+    assert expr.expression_name == "name"
+
+    assert expr.distrib == sp.stats.genextreme
+    assert not expr.is_distrib_discrete
+
+    assert expr.parameters_list == ["c", "loc", "scale"]
+
+    bounds = {"c": [-inf, inf], "loc": [-inf, inf], "scale": [0, inf]}
+    assert expr.boundaries_parameters == bounds
+
+    param_expr = {"loc": "c1+c2*pred1", "scale": "c3+c4*pred2**2", "c": "c5"}
+    assert expr.parameters_expressions == param_expr
+
+    coeffs = ["c1", "c2", "c3", "c4", "c5"]
+    assert expr.coefficients_list == coeffs
+
+    coeffs_per_param = {"loc": ["c1", "c2"], "scale": ["c3", "c4"], "c": ["c5"]}
+    assert expr.coefficients_dict == coeffs_per_param
+
+
+def test_expression_norm():
+
+    expression_str = "norm(loc=c1 + (c2 - c1) / ( 1 + np.exp(c3 * __GMT_t__ + c4 * __GMT_tm1__ - c5) ), scale=c6)"
+
+    expr = Expression(expression_str, expr_name="name")
+
+    assert expr.expression == expression_str
+    assert expr.expression_name == "name"
+
+    assert expr.distrib == sp.stats.norm
+    assert not expr.is_distrib_discrete
+
+    assert expr.parameters_list == ["loc", "scale"]
+
+    bounds = {"loc": [-inf, inf], "scale": [0, inf]}
+    assert expr.boundaries_parameters == bounds
+
+    param_expr = {"loc": "c1+(c2-c1)/(1+np.exp(c3*GMT_t+c4*GMT_tm1-c5))", "scale": "c6"}
+    assert expr.parameters_expressions == param_expr
+
+    coeffs = ["c1", "c2", "c3", "c4", "c5", "c6"]
+    assert expr.coefficients_list == coeffs
+
+    coeffs_per_param = {"loc": ["c1", "c2", "c3", "c4", "c5"], "scale": ["c6"]}
+    assert expr.coefficients_dict == coeffs_per_param
+
+
+def test_expression_binom():
+    # a discrete distribution
+
+    expression_str = "binom(loc=c1, n=5, p=7)"
+
+    expr = Expression(expression_str, expr_name="name")
+
+    assert expr.expression == expression_str
+    assert expr.expression_name == "name"
+
+    assert expr.distrib == sp.stats.binom
+    assert not expr.is_distrib_discrete
```

Comment:
```suggestion
    assert expr.is_distrib_discrete
```

Ah yes this is wrong. As it did not fail this reveals another small problem.